community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/StringSchemaKey.java

```diff
@@ -123,20 +126,10 @@ int compareValueTo( StringSchemaKey other )
             return 0;
         }
 
-        try
-        {
-            // TODO change to not throw
-            return codePointByteArrayCompare( bytes, other.bytes, ignoreLength || other.ignoreLength );
-        }
-        catch ( Exception e )
-        {
-            // We can not throw here because we will visit this method inside a pageCursor.shouldRetry() block.
-            // Just return a comparison that at least will be commutative.
-            return byteArrayCompare( bytes, other.bytes );
-        }
+        return byteArrayCompare( bytes, other.bytes, ignoreLength | other.ignoreLength );
```

Comment:
Not using `codePointArrayCompare` here worries me a bit since `UTF8StringValue` use code point comparison. Do we know that order will be the same as for rest of Neo?

Also, if we decide that this is fine, we can remove `ignoreLength` from `codePointByteArrayCompare` completely.

Reply:
All our tests involving range searches verifies order, comparing that order to that of the values module. So from that POV there are tests for it. Now is it exhaustive...?

And yes ignoreLength can be removed then, and also it can be made private again.